MetaMask/metamask-extension
# Updating Avatar components to forward refs

## Explanation
Currently the Avatar components don't forward refs so the root html element can't be accessed. This is an issue if the refs need to be used on the Avatar components for a popover. This PR forwards refs down to the Box component for all Avatar components.

The changes look more severe than they are because it looks like the entire component code has been replaced but really I am only changing 3 lines. The ones pictured below and the `displayName` which is required for a component that uses `forwardRef`

<img width="300" alt="Screenshot 2023-04-20 at 5 13 29 PM" src="https://user-images.githubusercontent.com/8112138/233296464-4307514f-6c11-48cb-91bf-46010bc10c0b.png">

Related to https://github.com/MetaMask/metamask-extension/pull/18676

## Screenshots/Screencaps



### Before
No visual changes to the Avatar components


https://user-images.githubusercontent.com/8112138/233295036-48610a1b-ce23-4601-8163-4901fb5b3248.mov

### After
No visual changes to the Avatar components


https://user-images.githubusercontent.com/8112138/233295130-67a44815-1adc-4ad3-80e1-ae7fd3cf535f.mov

## Manual Testing Steps

- I've written tests to check for refs so you could trust in the CI or
- Check out this branch
- Run `yarn jest ui/components/component-library` 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
